parallel-install/pkg/installation/installation_test.go

```diff
@@ -0,0 +1,352 @@
+				hc: hc,
+			}
+			overridesProvider := &mockOverridesProvider{}
+			eng := engine.NewEngine(overridesProvider, provider, "", engine.Config{
+				WorkersCount: 2,
+				Log:          log.Printf,
+			})
+
+			err := i.StartKymaInstallation(kubeClient, provider, overridesProvider, eng)
+
+			assert.Error(t, err)
+			assert.EqualError(t, err, "Kyma prerequisites installation failed due to the timeout")
+		})
+		t.Run("due to quit timeout", func(t *testing.T) {
+			kubeClient := fake.NewSimpleClientset()
+
+			hc := &mockHelmClient{
+				componentProcessingTime: 300,
+			}
+			provider := &mockProvider{
+				hc: hc,
+			}
+			overridesProvider := &mockOverridesProvider{}
+			eng := engine.NewEngine(overridesProvider, provider, "", engine.Config{
+				WorkersCount: 2,
+				Log:          log.Printf,
+			})
+
+			start := time.Now()
+			err := i.StartKymaInstallation(kubeClient, provider, overridesProvider, eng)
+			end := time.Now()
+
+			elapsed := end.Sub(start)
+
+			assert.Error(t, err)
+			assert.EqualError(t, err, "Force quit: Kyma prerequisites installation failed due to the timeout")
+
+			// One component installation lasts 300 ms
+			// Quit timeout occurs at 250 ms
+			// Check if program quits in the meantime
+			assert.GreaterOrEqual(t, elapsed.Milliseconds(), int64(250))
+			assert.Less(t, elapsed.Milliseconds(), int64(300))
+		})
+	})
+
+	t.Run("should install prerequisites and fail to install Kyma components", func(t *testing.T) {
+		t.Run("due to cancel timeout", func(t *testing.T) {
+			kubeClient := fake.NewSimpleClientset()
+
+			hc := &mockHelmClient{
+				componentProcessingTime: 40,
+			}
+			provider := &mockProvider{
+				hc: hc,
+			}
+			overridesProvider := &mockOverridesProvider{}
+			eng := engine.NewEngine(overridesProvider, provider, "", engine.Config{
+				WorkersCount: 2,
+				Log:          log.Printf,
+			})
+
+			err := i.StartKymaInstallation(kubeClient, provider, overridesProvider, eng)
+
+			assert.Error(t, err)
+			assert.EqualError(t, err, "Kyma installation failed due to the timeout")
+		})
+		t.Run("due to quit timeout", func(t *testing.T) {
+			kubeClient := fake.NewSimpleClientset()
+
+			inst := newInstallation()
+
+			// Changing it to higher amount to minimize difference between cancel and quit timeout
+			inst.Cfg.CancelTimeout = 220 * time.Millisecond
+
+			hc := &mockHelmClient{
+				componentProcessingTime: 70,
+			}
+			provider := &mockProvider{
+				hc: hc,
+			}
+			overridesProvider := &mockOverridesProvider{}
+			eng := engine.NewEngine(overridesProvider, provider, "", engine.Config{
+				WorkersCount: 2,
+				Log:          log.Printf,
+			})
+
+			start := time.Now()
+			err := inst.StartKymaInstallation(kubeClient, provider, overridesProvider, eng)
+			end := time.Now()
+
+			elapsed := end.Sub(start)
+
+			assert.Error(t, err)
+			assert.EqualError(t, err, "Force quit: Kyma installation failed due to the timeout")
+
+			// Prerequisites and two components installation lasts 300 ms
```

Comment:
3 prereq + 2 components * 71[ms] = 355[ms]. The comment mentions 300[ms]. Am I missing something?

Reply:
I am using 2 workers, so components are installed/uninstalled in parallel. I will add this as a comment, as it might be confusing in the future.